### PR TITLE
Week2 homework2 svetlana

### PR DESCRIPTION
**Pull request by svetikzp**

### CODE REVIEW

File: week2/project/ecommerce/src/App.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import HomePage from './pages/HomePage/HomePage';
+
+
+
+function App() {
+
+   return (
+      <div className="App">
+         <Routes>
+            <Route path='/' element={<HomePage />} />
```
Comment (ObadaMoustafa) [inline week2/project/ecommerce/src/App.js]: I checked the `HomePage` component and it's empty then if you don't need the user to use the endpoint of `/` you can modify this line to be 
```js
        <Route path="/" element={<Navigate to="/products" />} />
```

and don't forget to import `Navigate` from react router dom
✏️need to modify✏️

File: week2/project/ecommerce/src/App.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            <Route path='/products' element={<ProductsPage />} />
+            <Route path='/products/product/:itemId' element={<DetailProductPage />} />
```
Comment (ObadaMoustafa) [inline week2/project/ecommerce/src/App.js]: the path is only enough to be `/products/:itemId`
📝advise📝

File: week2/project/ecommerce/src/pages/ProductsPage/categories/categories.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+         {errorObj.isError && <ErrorMessage errorMsg={errorObj.message} />}
+         <div className='categories'>
+            {isLoading ? <div><Revs /></div> :
+               CategoriesList.map((category, index) => {
+                  return <Category key={index} category={category} onClick={onClick} activeCategory={activeCategory} />
+               })}
+         </div>
+      </div>
+   )
+}
```
Comment (ObadaMoustafa) [inline week2/project/ecommerce/src/pages/ProductsPage/categories/categories.jsx:40]: its better to separate the logic of error, loading and showing results
📝advise📝

File: week2/project/ecommerce/src/pages/ProductsPage/products/products.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import Product from './product'
+import React, { useState } from 'react'
```
Comment (ObadaMoustafa) [inline week2/project/ecommerce/src/pages/ProductsPage/products/products.jsx]: you can import `useEffect` here as well and use it directly in line 31
📝advise📝

**Review by ObadaMoustafa — APPROVED** [review]
wow Sevetlana well done the structure of folders and files is very clear also your code is clean.
no consoles and no commented codes and that's great.

you need to change only one thing and I find it an opportunity to tell you about it. you can find in the comment below. and the rest are only advises and not mandatory to modify to get approve.

again congratulations you did great 🚀🚀💥

File: week2/project/ecommerce/src/App.js (hunk 1 above)
Comment (svetikzp) [reply week2/project/ecommerce/src/App.js]: I changed the line.

File: week2/project/ecommerce/src/App.js (hunk 2 above)
Comment (svetikzp) [reply week2/project/ecommerce/src/App.js]: Fixed.

**Review by ObadaMoustafa — APPROVED** [review]
🚀🚀🚀

**Pull request closed without merging**